### PR TITLE
Export function to detect if wallet plugin is installed

### PR DESCRIPTION
**Pull request by Ben-Rey**

### CODE REVIEW

File: src/massaStation/MassaStationDiscovery.ts [hunk 1]
```diff
@@ -18,13 +18,29 @@ export async function isMassaStationInstalled(): Promise<boolean> {
...
   }
 
-  const isMassaStation = (module) =>
-    module.name === MS_WALLET_PLUGIN_NAME &&
-    module.author === MS_WALLET_PLUGIN_AUTHOR;
+  return !!response.result.find(
+    (module) =>
+      module.name === MS_WALLET_PLUGIN_NAME &&
+      module.author === MS_WALLET_PLUGIN_AUTHOR,
+  );
```
Comment (Thykof) [inline src/massaStation/MassaStationDiscovery.ts:38]:
```suggestion
  module.status === 'UP',
  );
```

Comment (Thykof) [reply src/massaStation/MassaStationDiscovery.ts:38]: the plugin wallet can be installed but Off

Comment (Thykof) [reply src/massaStation/MassaStationDiscovery.ts:39]: Property 'status' does not exist on type 'NodeModule'.

File: src/massaStation/MassaStationDiscovery.ts [hunk 2]
```diff
@@ -18,13 +18,34 @@ export async function isMassaStationInstalled(): Promise<boolean> {
...
   }
 
-  const isMassaStation = (module) =>
-    module.name === MS_WALLET_PLUGIN_NAME &&
-    module.author === MS_WALLET_PLUGIN_AUTHOR;
+  return !!response.result.find(
+    (module: PluginInfo) =>
+      module.name === MS_WALLET_PLUGIN_NAME &&
+      module.author === MS_WALLET_PLUGIN_AUTHOR &&
+      module.status === 'UP',
```
Comment (Thykof) [inline src/massaStation/MassaStationDiscovery.ts:38]:
```suggestion
      module.status === 'Up',
```

File: src/massaStation/MassaStationDiscovery.ts [hunk 3]
```diff
@@ -1,30 +1,55 @@
...
+  return Boolean(findWalletPlugin(response.result));
+}
 
+export async function isMassaWalletEnabled(): Promise<boolean> {
+  const response = await fetchPluginData();
+
+  if (response.isError) return false;
+
+  const walletPlugin = findWalletPlugin(response.result);
+  return walletPlugin && walletPlugin.status === 'UP';
```
Comment (Thykof) [inline src/massaStation/MassaStationDiscovery.ts:37]:
```suggestion
  return walletPlugin && walletPlugin.status === 'Up';
```

File: src/massaStation/MassaStationDiscovery.ts [hunk 4]
```diff
@@ -1,30 +1,55 @@
...
+}
+
+export async function isMassaStationAndWalletPluginInstalled(): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (Ben-Rey) [reply src/massaStation/MassaStationDiscovery.ts:40]: I need it to detect the provider

File: src/providersManager/providerList.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ export const providerList: ProviderList[] = [
   },
   {
     name: 'MASSA_STATION',
-    checkInstalled: isMassaStationInstalled,
+    checkInstalled: isMassaStationAndWalletPluginInstalled,
```
Comment (Thykof) [inline src/providersManager/providerList.ts:23]: we want the wallet plugin to be up

```suggestion
    checkInstalled: isMassaWalletEnabled,
```

File: src/massaStation/MassaStationDiscovery.ts [hunk 6]
```diff
@@ -1,30 +1,55 @@
...
+export async function isMassaWalletEnabled(): Promise<boolean> {
+  const response = await fetchPluginData();
+
+  if (response.isError) return false;
+
+  const walletPlugin = findWalletPlugin(response.result);
+  return walletPlugin && walletPlugin.status === 'Up';
+}
+
+export async function isMassaStationAndWalletPluginInstalled(): Promise<boolean> {
```
Comment (Thykof) [inline src/massaStation/MassaStationDiscovery.ts:40]: I don't think we need this function

File: src/massaStation/MassaStationDiscovery.ts [hunk 7]
```diff
@@ -1,30 +1,55 @@
...
+    (plugin) => plugin.name === PLUGIN_NAME && plugin.author === PLUGIN_AUTHOR,
   );
+}
+
+export async function isMassaStationAvailable(): Promise<boolean> {
+  const response = await fetchPluginData();
+  return !response.isError;
+}
+
+export async function isMassaWalletInstalled(): Promise<boolean> {
```
Comment (Thykof) [inline src/massaStation/MassaStationDiscovery.ts:25]: maybe we don't need this function

**Review by Thykof — COMMENTED** [review]
the most important I thing is to knwon if the wallet plugin is enable, and we also want to know if at least massa station is running

**Review by peterjah — APPROVED** [review]

**Pull request merged**